src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/WarehouseRequestResolver.java

```diff
@@ -48,222 +32,20 @@ public WarehouseRequestResolver(
-
-                //todo This works okay for now. In the future we want something better here though.
-
-                final BlockPos itemStackPos = wareHouse.getPositionOfChestWithItemStack(itemStack -> stack == itemStack);
-                final ILocation itemStackLocation =
-                  manager.getFactoryController().getNewInstance(TypeConstants.ILOCATION, itemStackPos, wareHouse.getWorld().getDimension().getType().getId());
-
-                final Delivery delivery = new Delivery(itemStackLocation, completedRequest.getRequester().getLocation(), deliveryStack.copy(), getDefaultDeliveryPriority(true));
-
-                final IToken<?> requestToken =
-                  manager.createRequest(new WarehouseRequestResolver(completedRequest.getRequester().getLocation(), completedRequest.getId()), delivery);
-
-                deliveries.add(manager.getRequestForToken(requestToken));
-                remainingCount -= ItemStackUtils.getSize(matchingStack);
-
-                if (remainingCount <= 0)
-                {
-                    break tileentities;
-                }
-            }
-        }
-
-        return deliveries.isEmpty() ? null : deliveries;
-    }
-
-    @Override
-    public void onAssignedRequestBeingCancelled(@NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> request)
-    {
-
-    }
-
-    @Override
-    public void onAssignedRequestCancelled(@NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> request)
-    {
-
-    }
-
-    private static Set<TileEntityWareHouse> getWareHousesInColony(final Colony colony)
-    {
-        return colony.getBuildingManager().getBuildings().values().stream()
-                 .filter(building -> building instanceof BuildingWareHouse)
-                 .map(building -> (TileEntityWareHouse) building.getTileEntity())
-                 .collect(Collectors.toSet());
-    }
-
-    @Override
-    public void onRequestedRequestComplete(@NotNull final IRequestManager manager, @NotNull final IRequest<?> request)
-    {
-    }
-
-    @Override
-    public void onRequestedRequestCancelled(@NotNull final IRequestManager manager, @NotNull final IRequest<?> request)
-    {
-    }
-
-    @NotNull
-    @Override
-    public ITextComponent getRequesterDisplayName(@NotNull final IRequestManager manager, @NotNull final IRequest<?> request)
-    {
-        return new TranslationTextComponent(TranslationConstants.COM_MINECOLONIES_BUILDING_WAREHOUSE_NAME);
+        return wareHouses.stream()
+                .anyMatch(wareHouse -> wareHouse.hasMatchingItemStackInWarehouse(itemStack -> requestToCheck.getRequest().matches(itemStack),
+                requestToCheck.getRequest().getMinimumCount()));
     }
 
     @Override
-    public int getPriority()
+    protected AbstractWarehouseRequestResolver newInstanceOfSelf(@NotNull final ILocation location, @NotNull final IToken<?> token)
```

Comment:
No this does not work.
This is completely hats on ass logic.

Why is this here!?!?!?

Reply:
This is gone.